toolset/benchmark/test_types/verifications.py

```diff
@@ -384,4 +393,42 @@ def verify_query_cases(self, cases, url, check_updates=False):
                     expected_len, headers, body, case_url, max_infraction)
                 problems += verify_headers(self.request_headers_and_body, headers, case_url)
 
+    # verify the number of queries and rows read for 20 queries, with a concurrency level of 512, on 10.000 http requests
+    problems+=verify_queries_count(self, "World", url+"20", 512, ab_queries_count, expected_queries, 20 * .99 * ab_queries_count)
+    return problems
+
+
+def verify_queries_count(self, tbl_name, url, concurrency=512, count=15000, expected_queries=15000, expected_rows=15000):
+    '''
+    Checks that the number of executed queries, at the given concurrency level, 
+    corresponds to: the total number of http requests made * the number of queries per request
+    '''
+    log("VERIFYING QUERIES COUNT FOR %s" % url,border='-', color=Fore.WHITE + Style.BRIGHT)
+    dbType=self.get_db_type()
+
+    problems = []
+    dbChecker=None
+
+    if dbType == "mysql":
```

Comment:
This as well. You can make a `verify_queries` method on the `Database` class inside each database folder and then call it here instead of checking for the proper db type. Something like:

```
from toolset.databases import databases

databases[dbType].verify_queries(self.config)
```

Reply:
I must necessarily add several methods in each `Database` class, only one `verify_queries` method would not be very clean (see https://github.com/TechEmpower/FrameworkBenchmarks/pull/5145/commits/87358fb0bc5c10d69c0d49c7e377f6c0341e35c5)

Reply:
I think he's saying make a new abstract method on the parent class called `verify_queries` and then each subtype would implement it their own way, so instead of
```Python
if dbType == "mysql":
  dbChecker=MysqlCheck(self.config)
elif dbType == "postgres":
  dbChecker=PostgresCheck(self.config, tbl_name)
elif dbType == "mongodb":
  dbChecker=MongoDbCheck(self.config)
```
you could just have what he wrote:
```Python
databases[dbType].verify_queries(self.config)
```
as `databases[dbType]` would be the subclass and therefore `verify_queries` would be the type-specific implementation - if `databases[dbType]` was a MySQL type, then it's implementation of `verify_queries` would be whatever `MysqlCheck(self.config)` does.

Reply:
ok, but `verify_queries` is common to all Database classes:
Overall, she does that:
- getQueries & getRows
- do `ab`
- getQueries & getRows
- compare queries and row count with expected values

These are the `getQueries` and `getRows` methods that are specific to a subclass

Reply:
`verify_queries` added in a Database super class: see https://github.com/TechEmpower/FrameworkBenchmarks/pull/5145/commits/a5c826a028c8cef31e8465af8cc695d1aeef494c